pkg/proc/proc_test.go

```diff
@@ -3714,3 +3714,28 @@ func TestDWZCompression(t *testing.T) {
 		}
 	})
 }
+
+func TestMapLoadConfigWithReslice(t *testing.T) {
+	// Check that load configuration is respected for resliced maps.
+	withTestProcess("testvariables2", t, func(p proc.Process, fixture protest.Fixture) {
+		zolotovLoadCfg := proc.LoadConfig{FollowPointers: true, MaxStructFields: -1, MaxVariableRecurse: 3, MaxStringLen: 10, MaxArrayValues: 10}
+		assertNoError(proc.Continue(p), t, "First Continue()")
+		scope, err := proc.GoroutineScope(p.CurrentThread())
+		assertNoError(err, t, "GoroutineScope")
+		m1, err := scope.EvalExpression("m1", zolotovLoadCfg)
+		assertNoError(err, t, "EvalVariable")
+		t.Logf("m1 returned children %d (%d)", len(m1.Children)/2, m1.Len)
+
+		expr := fmt.Sprintf("(*(*%q)(%d))[10:]", m1.DwarfType.String(), m1.Addr)
+		t.Logf("expr %q\n", expr)
+
+		m1cont, err := scope.EvalExpression(expr, zolotovLoadCfg)
+		assertNoError(err, t, "EvalVariable")
+
+		t.Logf("m1cont returned children %d", len(m1cont.Children)/2)
```

Comment:
Why log the length of the children divided by 2 here (and above) but assert the full value below?

Reply:
The full value would be 10 (the value of MaxArrayValues). But I forgot to divide by 2 in the call to Fatal.